addons/mail/static/src/models/thread/thread.js

```diff
@@ -1385,8 +1388,11 @@ function factory(dependencies) {
             if (this.channel_type === 'chat' && this.correspondent) {
                 return this.custom_channel_name || this.correspondent.nameOrDisplayName;
             }
-            if (this.channel_type === 'group') {
-                return this.name || this.members.map(partner => partner.nameOrDisplayName).join(', ');
+            if (this.channel_type === 'group' && !this.name) {
+                const partners = this.members.map(partner => partner.nameOrDisplayName).sort();
+                const guests = this.guestMembers.map(guest => guest.name).sort();
+                const groupName = partners.concat(guests).join(', ');
+                return groupName;
```

Comment:
The `groupName` variable is not needed. Also, while you're at it, I'd recommend using the spread operator to concatenate arrays. It's less noisy in my opinion and feels more like modern JavaScript.
```suggestion
                return [...partnerNames, ...guestNames].join(", ");
```